### DISCUSSION TITLE
GC9A01 on Teensy 4.1 SPI1 instead of SPI0

### DISCUSSION THREAD

**Discussion by klaasdc** [Bodmer/TFT_eSPI#2649] [Compatible displays and setup files]
Hi,

I'd like to try experimenting with this library. I use PlatformIO and have a GC9A01 connected to SPI1 on a Teensy 4.1

```
#define TFT_MISO -1
#define TFT_MOSI 26
#define TFT_SCLK 27
#define TFT_CS   0  // Chip select control pin
#define TFT_DC   28  // Data Command control pin
#define TFT_RST  29  // Reset pin (could connect to RST pin)
```

I tried `#define TFT_SPI_PORT 1` but that gives a compilation error:

```
In file included from .pio\libdeps\teensy41\TFT_eSPI/User_Setup_Select.h:30:0,
                 from .pio\libdeps\teensy41\TFT_eSPI\TFT_eSPI.h:68,
                 from .pio\libdeps\teensy41\TFT_eSPI\TFT_eSPI.cpp:16:
.pio\libdeps\teensy41\TFT_eSPI/User_Setup.h:270:22: error: invalid initialization of non-const reference of type 'SPIClass&' from an rvalue of type 'int'
```
Is it possible to have it work on SPI1 instead?

**Comment by Bodmer** [comment]
The Teensy 4.1 is treated as a generic processor, so there is no optimisation or SPI port configuration options. In practice the Teensy board package SPi code is well written and performant so there are no plans to add any optimisation.This means the Arduino board package defaults are used for the SPI port.

Modification of the library code will be necessary to use other port numbers.

**Comment by klaasdc** [comment]
@Bodmer Thanks for the reply.
I thought the SPI port selection was related to `#define TFT_SPI_PORT 1` inside User_Setup.h
But eventually all works after changing line 13 in TFT_eSPI_Generic.h from SPI to SPI1

**Reply by Bodmer** [reply]
OK, it would be easy to add that option for the Teensy, I will add to the TODO list.
